controllers/handler/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -258,13 +258,19 @@ func (r *NodeNetworkConfigurationPolicyReconciler) SetupWithManager(mgr ctrl.Man
 
 	// Reconcile all NNCPs if Node is updated (for example labels are changed), node creation event
 	// is not needed since all NNCPs are going to be Reconcile at node startup.
-	err = ctrl.NewControllerManagedBy(mgr).
-		For(&corev1.Node{}).
-		Watches(&source.Kind{Type: &corev1.Node{}}, handler.EnqueueRequestsFromMapFunc(allPolicies), builder.WithPredicates(onLabelsUpdatedForThisNode)).
-		Complete(r)
+	c, err := controller.New("node_labels", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return errors.Wrap(err, "failed to create node_labels controller to NNCP Reconciler watching node label changes")
+	}
+	err = c.Watch(
+		&source.Kind{Type: &corev1.Node{}},
+		handler.EnqueueRequestsFromMapFunc(allPolicies),
+		onLabelsUpdatedForThisNode,
+	)
```

Comment:
We don't need `Complete(r)` ?

Reply:
We're creating the controller with `controller.New` on line, so we don't need Complete, as complete is used to instantiate the controller from the controller builder